### PR TITLE
Fix "alpha must be between 0 and 255"

### PR DESCRIPTION
**Pull request by JorisBodin**

### CODE REVIEW

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 1]
```diff
@@ -352,7 +357,7 @@ class ThreadFragment : Fragment() {
     }
 
     private companion object {
-        const val COLLAPSE_TITLE_THRESHOLD = 0.5
+        const val COLLAPSE_TITLE_THRESHOLD = 0.666
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:360]: Why is this threshold change better or needed ?

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 2]
```diff
@@ -117,12 +118,16 @@ class ThreadFragment : Fragment() {
 
         val defaultTextColor = context.getColor(R.color.primaryTextColor)
         appBar.addOnOffsetChangedListener { appBarLayout, verticalOffset ->
-            val total = appBarLayout.height * COLLAPSE_TITLE_THRESHOLD
-            val removed = appBarLayout.height - total
-            val progress = (((-verticalOffset.toFloat()) - removed).coerceAtLeast(0.0) / total).toFloat() // Between 0 and 1
-            val opacity = (progress * 255).roundToInt()
 
+            val subjectHeight = appBarLayout.height.toDouble()
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:122]: This `toDouble()` is not needed as both other lines will still be double due to computation always involving a double. Do we really want it?

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 3]
```diff
@@ -117,12 +118,16 @@ class ThreadFragment : Fragment() {
...
+            val impactingHeight = subjectHeight * COLLAPSE_TITLE_THRESHOLD
+            val nonImpactingHeight = subjectHeight - impactingHeight
+
+            val absoluteProgress = verticalOffset.absoluteValue.toDouble() - nonImpactingHeight
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:126]: This `toDouble()` is also not needed, same question

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:360]: With `0.5`, there is some frames where the subject is totally hidden.
It feels more smooth this way.
image

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:122]: For the sake of clarity, I prefer to write it this way.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 3 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:126]: https://github.com/Infomaniak/android-kMail/pull/881#discussion_r1172454057

**Review by LunarX — APPROVED** [review]

**Pull request merged**